lib/api/CDataFrameAnalysisInstrumentation.cc

```diff
@@ -64,10 +69,32 @@ const std::string PROGRESS_PERCENT{"progress_percent"};
 
 const std::size_t MAXIMUM_FRACTIONAL_PROGRESS{std::size_t{1}
                                               << ((sizeof(std::size_t) - 2) * 8)};
+
+std::string bytesToString(double value) {
+    std::ostringstream stream;
+    stream << std::fixed;
+    stream << std::setprecision(3);
+    value /= 1024;
+    if (value < 1000) {
+        stream << value;
+        stream << " kb";
+        return stream.str();
+    }
+    value /= 1024;
+    stream << value;
+    stream << " mb";
+    return stream.str();
+}
+
+std::string bytesToString(std::int64_t bytes) {
+    return bytesToString(static_cast<double>(bytes));
+}
```

Comment:
👍 . (Almost feels like this is worth factoring out and putting somewhere. I'm sure there are other cases where we want to print memory usage. Not sure where this belongs though so something for another change.)